packages/react-vanilla-components/src/utils/withRuleEngine.tsx

```diff
@@ -0,0 +1,101 @@
+
+export const richTextString = (stringMsg = '') => {
+  const htmlProp = { __html: sanitizeHTML(stringMsg) };
+  return (<div dangerouslySetInnerHTML={htmlProp} />);
+};
+const formateErrorMessage = (state: FieldViewState) => {
+  const errorMessage = state.errorMessage === '' && state.valid === false ? DEFAULT_ERROR_MESSAGE : state.errorMessage;
+  return errorMessage;
+};
+const getValidationState = (state: FieldViewState) => {
+  const optionalValidation = !checkIfConstraintsArePresent(state) && state.type === 'string';
+  const validationState = state.valid === false ? 'invalid' : ((state.valid === undefined || isEmpty(state.value) || optionalValidation) ? undefined : 'valid');
+  return validationState;
+};
+const getLocalizeLabel = (i18n: any, state: FieldViewState, richText: boolean | undefined) => {
+  const labelId = getOrElse(state, ['properties', 'afs:translationIds', 'label.value']);
+  const label = labelId ? i18n.formatMessage({ id: labelId, defaultMessage: state?.label?.value }) : state?.label?.value;
+  return richText ? richTextString(label) : label;
+};
+const getLocalizeDescription = (i18n: any, state: FieldViewState) => {
+  const descriptionId = getOrElse(state, ['properties', 'afs:translationIds', 'description']);
+  const description = descriptionId ? i18n.formatMessage({ id: descriptionId, defaultMessage: state?.description }) : state?.description;
+  return description ? richTextString(description) : null;
+};
+const getLocalizePlaceholder = (i18n: any, state: FieldViewState) => {
+  const placeholderId = getOrElse(state, ['properties', 'afs:translationIds', 'placeholder']);
+  const placeholder = placeholderId ? i18n.formatMessage({ id: placeholderId, defaultMessage: state?.placeholder }) : state?.placeholder;
+  return placeholder;
+};
+
+const getToolTip = (state: FieldViewState) => {
+  return state.tooltip ? richTextString(state.tooltip) : null;
+};
+export function withRuleEngine(Component: JSXElementConstructor<any>) {
+  return function WrappedComponent(fieldset: State<FieldJson>) {
+    const [state, handlers] = useRuleEngine(fieldset);
+    const i18n = useFormIntl();
+    const localizeState = {
+      ...state,
+      description: getLocalizeDescription(i18n, state),
+      placeholder: getLocalizePlaceholder(i18n, state),
+      tooltip: getToolTip(state),
+      label: {
+        ...state?.label,
+        value: getLocalizeLabel(i18n, state, state?.label?.richText),
+        visible: state.label?.visible !== false
+      },
+      isError: getValidationState(state) === 'invalid',
+      errorMessage: formateErrorMessage(state),
+      layout: {
+        orientation: 'horizontal',
+        ...(getOrElse(state, ['properties', 'afs:layout'], {}))
+      }
+    };
+    const visible = typeof state.visible === 'undefined' || state.visible;
+    // @ts-ignore
+    return visible ? <div className='field'><Component {...localizeState} {...handlers} /></div> : null;
```

Comment:
Why do you need this, this is not as per BEM ?

Reply:
Removed!